pluginframework/common/go_to_tf.go

```diff
@@ -0,0 +1,221 @@
+		} else {
+			destField.Set(reflect.ValueOf(types.Int64Null()))
+		}
+	case reflect.Float64:
+		// convert any kind of float to float64
+		float64Val := srcField.Convert(reflect.TypeOf(float64(0))).Interface().(float64)
+		// check if the value is non-zero or if the field is in the forceSendFields list
+		if float64Val != 0 || forceSendField {
+			destField.Set(reflect.ValueOf(types.Float64Value(float64Val)))
+		} else {
+			destField.Set(reflect.ValueOf(types.Float64Null()))
+		}
+	case reflect.String:
+		var strVal string
+		if srcField.Type().Name() != "string" {
+			// This case is for Enum Types.
+			strVal = getStringFromEnum(srcField)
+		} else {
+			strVal = srcFieldValue.(string)
+		}
+		// check if the value is non-zero or if the field is in the forceSendFields list
+		if strVal != "" || forceSendField {
+			destField.Set(reflect.ValueOf(types.StringValue(strVal)))
+		} else {
+			destField.Set(reflect.ValueOf(types.StringNull()))
+		}
+	case reflect.Struct:
+		if srcField.IsZero() {
+			// Skip zeros
+			return nil
+		}
+		// resolve the nested struct by recursively calling the function
+		if GoSdkToTfSdkStruct(srcFieldValue, destField.Addr().Interface(), ctx).ErrorsCount() > 0 {
+			panic("Error converting gosdk to tfsdk struct")
+		}
+	case reflect.Slice:
+		if srcField.IsNil() {
+			// Skip nils
+			return nil
+		}
+		destSlice := reflect.MakeSlice(destField.Type(), srcField.Len(), srcField.Cap())
+		for j := 0; j < srcField.Len(); j++ {
+
+			srcElem := srcField.Index(j)
+
+			destElem := destSlice.Index(j)
+			if err := goSdkToTfSdkSingleField(srcElem, destElem, true, ctx); err != nil {
+				return err
+			}
+		}
+		destField.Set(destSlice)
+	case reflect.Map:
+		if srcField.IsNil() {
+			// Skip nils
+			return nil
+		}
+		destMap := reflect.MakeMap(destField.Type())
+		for _, key := range srcField.MapKeys() {
+			srcMapValue := srcField.MapIndex(key)
+			destMapValue := reflect.New(destField.Type().Elem()).Elem()
+			destMapKey := reflect.ValueOf(key.Interface())
+			if err := goSdkToTfSdkSingleField(srcMapValue, destMapValue, true, ctx); err != nil {
+				return err
+			}
+			destMap.SetMapIndex(destMapKey, destMapValue)
+		}
+		destField.Set(destMap)
+	default:
+		panic(fmt.Errorf("unknown type for field: %s", srcField.Type().Name()))
+	}
+	return nil
+}
+
+func getStringFromEnum(srcField reflect.Value) string {
+	var stringMethod reflect.Value
+	if srcField.CanAddr() {
+		stringMethod = srcField.Addr().MethodByName("String")
+	} else {
+		// Create a new addressable variable to call the String method
+		addr := reflect.New(srcField.Type()).Elem()
+		addr.Set(srcField)
+		stringMethod = addr.Addr().MethodByName("String")
+	}
+	if stringMethod.IsValid() {
+		stringResult := stringMethod.Call(nil)
+		if len(stringResult) == 1 {
+			return stringResult[0].Interface().(string)
+		} else {
+			panic("num get string has more than one result")
+		}
+	} else {
+		panic("enum does not have valid .String() method")
+	}
+}
```

Comment:
See comment [here](https://github.com/databricks/terraform-provider-databricks/pull/3893/files#r1721764318) for why type casting doesn't necessarily make the code simpler

Reply:
Did we ever figure out why this is?

Reply:
Discussed offline and added comment, this case where `CanAddr()` returns false happens in unit tests due to the value being a constant